index.d.ts

```diff
@@ -0,0 +1,3 @@
+export default class MarkdownToolbarElement extends HTMLElement {
+  readonly field?: HTMLTextAreaElement;
```

Comment:
Why `readonly: field?: TextAreaElement` here instead of `readyonly field: HTMLTextAreaElement | undefined` like https://github.com/github/remote-input-element/pull/6/files#diff-b52768974e6bc0faccb7d4b75b162c99R2?

I also noticed that 

```js
import MarkdownToolbarElement from '.'
const element = document.querySelector('*')

if (element instanceof MarkdownToolbarElement) {
  element.field.value = element.field.value.trim()
}
```

does not fail TS check but it does fail in Flow. Is this expected?

Reply:
So I thought initially that the only way to do nullable types was by using the union `| undefined`  but I've since found out that appending `?` to the argument is a shorthand of `| undefined`:

Ref: https://www.typescriptlang.org/docs/handbook/advanced-types.html#optional-parameters-and-properties

You do need to set `strictNullChecks` though.

Reply:
>does not fail TS check but it does fail in Flow. Is this expected?

How are you running the checks here? I don't get a error in either flow or typescript :/

Reply:
I tested TS by adding a `test.ts` in the same directory. I tested Flow in `github/github` `responsive-markdown-toolbar.js` since it's a module import definition. 

> So I thought initially that the only way to do nullable types was by using the union `| undefined` but I've since found out that appending `?` to the argument is a shorthand of `| undefined`:

I think we should be consistent across modules or we should have specific reasons to choose one over the other.

https://github.com/github/markdown-toolbar-element/blob/9b21198917edf0b76e0f394770a9ef4ca1ad8031/index.js#L218-L224

In this case we might actually return `HTMLTextAreaElement | undefined | null`.

Reply:
> I think we should be consistent across modules or we should have specific reasons to choose one over the other.

Sounds good. I think we should use the more descriptibe `| undefined | null` everywhere since we don't know what `strictNullChecks` will be set to in the application that consumes the element and so we can return `null` like you mentioned.